pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -66,12 +66,13 @@ This file implements string parsing and creation for NumPy datetime.
  *
  * Returns 0 on success, -1 on failure.
  */
-int parse_iso_8601_datetime(char *str, int len,
+int parse_iso_8601_datetime(const char *str, int len,
                             npy_datetimestruct *out,
                             int *out_local, int *out_tzoffset) {
     int year_leap = 0;
     int i, numdigits;
-    char *substr, sublen;
+    const char *substr;
+    int sublen;
```

Comment:
was this wrong as char before?

Reply:
Because it was being assigned an `int` :) So a potential for an overflow or underflow if string to be parsed is longer than 127 characters.

Reply:
A `int` variable is assigned to `sublen`  in this function. So I think it was wrong.

Reply:
Great.  Can you add a test case for the overflow/underflow?

Reply:
Test is added.

Reply:
Link to test implementation (test fails on current master due to overflow): https://github.com/pandas-dev/pandas/pull/25895/files#diff-e295b05df69c00ad3e6d6d4d97360cccR65